### PR TITLE
sql implementation attempt

### PR DESCRIPTION
**Pull request by michimau**
For my business case i need to run a sql statement against a generic org dataset, since "where" is not good enough.

I hope what i have written makes some sense as I have no clue on how to compile the library -> hence untested :o)

### CODE REVIEW

**Comment by fxprunayre** [comment]
> I have no clue on how to compile the library

Ciao Mauro, you can simply click on CTRL + SHIFT + F3 to make the studio recompile all javajet files. Restarting the studio should also work but takes more time.

File: main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_begin.javajet [hunk 1]
```diff
@@ -87,6 +88,11 @@ oRing_<%=cid %>.AddPoint(xmin_<%=cid %>, ymin_<%=cid %>);
 poSpatialFilter_<%=cid %> = new org.gdal.ogr.Geometry(org.gdal.ogr.ogrConstants.wkbPolygon);
 poSpatialFilter_<%=cid %>.AddGeometry(oRing_<%=cid %>);
 
+if (!where.equals("") && !sql.equals("")) {
+    System.out.print("FAILURE:\n" + "You cannot use both where and sql.";
```
Comment (fxprunayre) [inline main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_begin.javajet]: Missing closing parenthesis here

File: main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_begin.javajet [hunk 2]
```diff
@@ -31,6 +31,7 @@ if ((metadatas != null) && (metadatas.size() > 0)) {
         String ymax = ElementParameterParser.getValue(node,"__YMAX__");
         String where = ElementParameterParser.getValue(node,"__WHERE__");
         String sql = ElementParameterParser.getValue(node,"__SQL__");
+        boolean dialect = ("true").equals(ElementParameterParser.getValue(node,"__DIALECT__"));
```
Comment (fxprunayre) [inline main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_begin.javajet]: ```"true".equals``` will work fine also

File: main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_java.xml [hunk 3]
```diff
@@ -57,10 +57,13 @@
 			NUM_ROW="8">
 			<DEFAULT>""</DEFAULT>
 		</PARAMETER>
-		<!-- TODO <PARAMETER NAME="SQL" FIELD="TEXT" REQUIRED="true"
+		<PARAMETER NAME="DIALECT" FIELD="CHECK" NUM_ROW="9">
+            <DEFAULT>false</DEFAULT>
+        </PARAMETER>
```
Comment (fxprunayre) [inline main/plugins/org.talend.sdi.designer.components/components/sOgrInput/sOgrInput_java.xml]: Cosmetic: maybe indent properly

**Comment by michimau** [comment]
I have reworked the "sOgrInput" component so it does support sql and sql dialect. 
I have tested and it is working. Please have a look.

**Comment by michimau** [comment]
screenshot from 2017-10-05 09-18-19

Both sql and where can be used together as in the example.

**Review by fxprunayre — APPROVED** [review]

**Pull request merged**